tests/basic_test.go

```diff
@@ -96,6 +97,87 @@ func testPolicyWithCachedCapturesAndNoDesiredStateRef(t *testing.T) {
 	})
 }
 
+func testPolicyWithFilterCaptureAndNoDesiredState(t *testing.T) {
+	t.Run("with a eqfilter capture expression and desired state that has no ref", func(t *testing.T) {
+		stateData := []byte(`
+routes:
+  running:
+  - destination: 0.0.0.0/0
+    next-hop-address: 192.168.100.1
+    next-hop-interface: eth1
+    table-id: 254
+  - destination: 1.1.1.0/24
+    next-hop-address: 192.168.100.1
+    next-hop-interface: eth1
+    table-id: 254
+  config:
+  - destination: 0.0.0.0/0
+    next-hop-address: 192.168.100.1
+    next-hop-interface: eth1
+    table-id: 254
+  - destination: 1.1.1.0/24
+    next-hop-address: 192.168.100.1
+    next-hop-interface: eth1
+    table-id: 254
+interfaces:
+  - name: eth1
+    type: ethernet
+    state: up
+    ipv4:
+      address:
+      - ip: 10.244.0.1
+        prefix-length: 24
+      - ip: 169.254.1.0
+        prefix-length: 16
+      dhcp: false
+      enabled: true
+  - name: eth2
+    type: ethernet
+    state: down
+    ipv4:
+      address:
+      - ip: 1.2.3.4
+        prefix-length: 24
+      dhcp: false
+      enabled: false
+`)
+		const capID0 = "cap0"
+		policySpec := types.PolicySpec{
+			Capture: map[string]string{
+				capID0: `routes.running.destination=="0.0.0.0/0"`,
+			},
+			DesiredState: stateData,
+		}
+
+		beforeGenerate := time.Now()
+		sleepTime := time.Second
+		time.Sleep(sleepTime)
+
+		cacheState := types.CachedState{}
+		obtained, err := nmpolicy.GenerateState(
+			policySpec,
+			stateData,
+			cacheState)
+
+		assert.NoError(t, err)
+		expectedCapID0 := `
+routes:
+  running:
+  - destination: 0.0.0.0/0
+    next-hop-address: 192.168.100.1
+    next-hop-interface: eth1
+    table-id: 254
+`
+		assert.Equal(t, "0", obtained.MetaInfo.Version)
+		assert.GreaterOrEqual(t, obtained.MetaInfo.TimeStamp.Sub(beforeGenerate), sleepTime)
+		assert.Len(t, obtained.Cache.Capture, 1)
+		assert.Contains(t, obtained.Cache.Capture, capID0)
+		assert.Equal(t, "", obtained.Cache.Capture[capID0].MetaInfo.Version)
+		assert.GreaterOrEqual(t, obtained.Cache.Capture[capID0].MetaInfo.TimeStamp.Sub(beforeGenerate), sleepTime)
+		assert.YAMLEq(t, expectedCapID0, string(obtained.Cache.Capture[capID0].State))
```

Comment:
Please perform one single assert on all the returning state.
This way of asserting is problematic, it does not show the whole picture when something is wrong (e.g. with the len assert, we see more captures than expected, but we do not see what is the content).

Reply:
We can't do that, we don't want to be strict with the yaml string, since two valid yamls can have different strings, maybe we can canonalize the obtained yamls so they looks always the same.

Reply:
Could you please explain this? Don't we know what to exactly get with just one exception*?

* The time stamp.

Reply:
The format yaml ensure that expected and obtained YAMLs contains exactly the same []byte if they have marshal the same content.